packages/ws-worker/src/mock/sockets.ts

```diff
@@ -74,39 +74,48 @@ export const mockChannel = (
 
 type ChannelMap = Record<string, ReturnType<typeof mockChannel>>;
 
-export const mockSocket = (
-  _endpoint?: string,
-  channels?: ChannelMap,
-  connect: () => Promise<void> = async () => {}
-) => {
-  const allChannels: ChannelMap = channels || {};
+export class MockSocket {
```

Comment:
I appreciate that this branch is a work in progress but is this a necessary change? Should it be branched out differently to the ava/loader thing?

Reply:
It was necessary given that. it was initially defined using an arrow function instead of a normal classic function which utilises `this` to grant it a constructor. 
It was then failing in v20 because there were several calls to it using the `new` operator.